File: src/main/java/me/stevemmmmm/server/game/enchants/EnchantProperty.java
package me.stevemmmmm.server.game.enchants;

public class EnchantProperty<T> {
    private T[] values;

    @SafeVarargs
    public EnchantProperty(T... values) {
        if (values.length > 3)
            throw new IllegalArgumentException("Enchants only have 3 levels. You had " + values.length + "properties.");

        this.values = values;
    }

    public T getValueAtLevel(int level) {
        return values[level - 1];
    }
}

File: src/main/java/com/thebluehats/server/game/managers/enchants/EnchantProperty.kt
package com.thebluehats.server.game.managers.enchants

class EnchantProperty<T> @SafeVarargs constructor(vararg values: T) {
    private val values: Array<out T>

    fun getValueAtLevel(level: Int): T {
        return values[level - 1]
    }

    init {
        require(values.size <= 3) { "Enchants only have 3 levels. You had " + values.size + "properties." }

        this.values = values
    }
}